packages/dds/tree/src/test/feature-libraries/modular-schema/comparison.spec.ts

```diff
@@ -152,15 +154,65 @@ describe("Schema Comparison", () => {
 		const identifierField = fieldSchema(FieldKinds.identifier, [emptyTree.name]);
 		const sequenceField = fieldSchema(FieldKinds.sequence, [emptyTree.name]);
 		const compare = (a: TreeFieldStoredSchema, b: TreeFieldStoredSchema): boolean =>
-			allowsFieldSuperset(defaultSchemaPolicy, repo, a, b);
+			allowsFieldSuperset(defaultSchemaPolicy, repo, a, b, false);
 		testOrder(compare, [neverField, storedEmptyFieldSchema, fieldOptionalEmptyTree]);
 		testOrder(compare, [neverField, fieldRequiredEmptyTree]);
+		testOrder(compare, [fieldRequiredEmptyTree, fieldOptionalEmptyTree, sequenceField]);
+		testOrder(compare, [identifierField, fieldOptionalEmptyTree, sequenceField]);
+		assert.equal(
+			getOrdering(fieldRequiredEmptyTree, storedEmptyFieldSchema, compare),
+			Ordering.Incomparable,
+		);
+		assert.equal(
+			getOrdering(identifierField, fieldRequiredEmptyTree, compare),
+			Ordering.Equal,
+		);
+		testPartialOrder(
+			compare,
+			[
+				neverField,
+				neverField2,
+				storedEmptyFieldSchema,
+				fieldRequiredEmptyTree,
+				fieldRequiredEmptyTree,
+				sequenceField,
+				identifierField,
+			],
+			[
+				[neverField, neverField2],
+				[identifierField, fieldRequiredEmptyTree],
+			],
+		);
+	});
+
+	it("allowsFieldSuperset-monotonic", () => {
+		const repo = new TreeStoredSchemaRepository();
+		updateTreeSchema(repo, brand("never"), neverTree);
+		updateTreeSchema(repo, emptyTree.name, emptyTree.schema);
+		const neverField2: TreeFieldStoredSchema = fieldSchema(FieldKinds.required, [
+			brand("never"),
+		]);
+		const identifierField = fieldSchema(FieldKinds.identifier, [emptyTree.name]);
+		const sequenceField = fieldSchema(FieldKinds.sequence, [emptyTree.name]);
+		const compare = (a: TreeFieldStoredSchema, b: TreeFieldStoredSchema): boolean =>
+			allowsFieldSuperset(defaultSchemaPolicy, repo, a, b);
+		testOrder(compare, [storedEmptyFieldSchema, fieldOptionalEmptyTree]);
+		testOrder(compare, [neverField, fieldRequiredEmptyTree]);
 		testOrder(compare, [
 			identifierField,
 			fieldRequiredEmptyTree,
 			fieldOptionalEmptyTree,
 			sequenceField,
 		]);
+		assert.equal(getOrdering(neverField, neverField2, compare), Ordering.Superset);
```

Comment:
previously these two were considered equal (which was validated by the old test, and still validated by the non-monotonic case). This could allow an infinite update loop between these two schema.

That said, that update loop required the schema to be "never schema" which are schema which are structurally impossible to have any nodes which are in schema for them, so not actually useful schema.